extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasMemory.cs

```diff
@@ -0,0 +1,351 @@
+        return collection.DeleteOneAsync(x => x.Id == record.Id, cancellationToken: cancellationToken);
+    }
+
+    public async IAsyncEnumerable<MemoryRecord> GetListAsync(
+        string index,
+        ICollection<MemoryFilter>? filters = null,
+        int limit = 1,
+        bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (limit <= 0)
+        {
+            limit = 10;
+        }
+
+        // Need to create a query and execute it without using $vector
+        var collection = this.GetCollectionFromIndexName(index);
+        var finalFilter = this.TranslateFilters(filters, index);
+
+        // We need to perform a simple query without using vector search
+        var cursor = await collection.FindAsync(finalFilter, cancellationToken: cancellationToken).ConfigureAwait(false);
+        var documents = await cursor.ToListAsync(cancellationToken).ConfigureAwait(false);
+
+        foreach (var document in documents)
+        {
+            var memoryRecord = FromMongodbMemoryRecord(document, withEmbeddings);
+
+            yield return memoryRecord;
+        }
+    }
+
+    public async IAsyncEnumerable<(MemoryRecord, double)> GetSimilarListAsync(
+        string index,
+        string text,
+        ICollection<MemoryFilter>? filters = null,
+        double minRelevance = 0,
+        int limit = 1,
+        bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (limit <= 0)
+        {
+            limit = 10;
+        }
+
+        // Need to create a search query and execute it
+        var collectionName = this.GetCollectionName(index);
+        var embeddings = await this._embeddingGenerator.GenerateEmbeddingAsync(text, cancellationToken).ConfigureAwait(false);
+
+        // Define vector embeddings to search
+        var vector = embeddings.Data.Span.ToArray();
+
+        // Need to create the filters
+        var finalFilter = this.TranslateFilters(filters, index);
+
+        var options = new VectorSearchOptions<MongoDbAtlasMemoryRecord>()
+        {
+            IndexName = this._utils.GetIndexName(collectionName),
+            NumberOfCandidates = limit,
+            Filter = finalFilter
+        };
+        var collection = this.GetCollectionFromIndexName(index);
+
+        // Run query
+        var documents = await collection.Aggregate()
+            .VectorSearch(m => m.Embedding, vector, limit, options)
+            .ToListAsync(cancellationToken).ConfigureAwait(false);
+
+        // If you check documentation Atlas normalize the score with formula
+        // score = (1 + cosine/dot_product(v1,v2)) / 2
+        // Thus it does not output the real cosine similarity, this is annoying so we
+        // need to recompute cosine similarity using the embeddings.
+#if DEBUG
+        //List<(double, double, double)> similarities = new();
+#endif
```

Comment:
could you remove the code commented out?

Looks like there are a bunch of #DEBUG blocks that could be removed across the file